framework/db/mssql/QueryBuilder.php

```diff
@@ -250,13 +250,45 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
         return $command;
     }
 
+    protected function buildAddCommentSql($comment, $table, $column = null)
+    {
+        $tableSchema = $this->db->schema->getTableSchema($table);
+
+        $schemaName = $tableSchema->schemaName ? "N'" . $tableSchema->schemaName . "'": 'SCHEMA_NAME()';
+        $tableName = "N" . $this->db->quoteValue($tableSchema->name);
+        $columnName = $column ? "N" . $this->db->quoteValue($column) : null;
+        $comment = "N" . $this->db->quoteValue($comment);
+
+        $functionParams = "
+            @name = N'MS_description',
+            @value = $comment,
+            @level0type = N'SCHEMA', @level0name = $schemaName,
+            @level1type = N'TABLE', @level1name = $tableName"
+            . ($column ? ", @level2type = N'COLUMN', @level2name = $columnName" : '') . ';';
+
+        return "
+            IF NOT EXISTS (
+                    SELECT 1
+                    FROM fn_listextendedproperty (
+                        N'MS_description',
+                        'SCHEMA', $schemaName,
+                        'TABLE', $tableName,
+                        " . ($column ? "'COLUMN', $columnName " : ' DEFAULT, DEFAULT ') . "
+                    )
+            )
+                EXEC sys.sp_addextendedproperty $functionParams
+            ELSE
+                EXEC sys.sp_updateextendedproperty $functionParams
```

Comment:
What is the purpose of having sp_updateextendedproperty? If you need to update comment you can always delete old comment and add new one. Add and delete is enough for all use cases. No need for update I think.

Reply:
I can revert the question to you and we are at the same point. So checking and updating or droping and inserting we will have at least two commands anyway. But since we can't drop without check before because it fails, we would need 3 queries: check, drop and insert. Or maybe I just could not see what is your point with this question, sorry.